### DISCUSSION TITLE
[Explore]set default aggregate based on selected column type

### DISCUSSION THREAD

**Discussion by mistercrunch** [apache/superset#18449] [Ideas]
We could save 2 clicks by assuming people want to `SUM` numeric columns and `COUNT(DISTINCT {col})` strings.

<img width="364" alt="Screen Shot 2021-02-18 at 2 51 30 PM" src="https://user-images.githubusercontent.com/487433/108432218-e1429f80-71f8-11eb-9f23-5413e47aff4d.png">

**Comment by junlincc** [comment]
@mistercrunch This popover is gonna function quite differently in d&d,  we will take it into consideration while redesigning.

**Comment by junlincc** [comment]
set dynamic aggregate function default based on column types
numeric and contains `_id_ _key_  _code_ ` -> COUNT(DISTINCT)
other numeric -> `SUM`
strings -> `COUNT(DISTINCT)`
TEXT -> SUM(LENGTH) (?)

@zhaoyongjie @villebro
